server/src/main/java/org/tctalent/server/service/db/impl/UserServiceImpl.java

```diff
@@ -610,6 +615,14 @@ public void updatePassword(UpdateUserPasswordRequest request) {
         String passwordEnc = passwordHelper.validateAndEncodePassword(request.getPassword());
         user.setPasswordEnc(passwordEnc);
         userRepository.save(user);
+
+        /* Update changePassword */
+        if (user.getCandidate().isChangePassword()) {
+            Candidate candidate = candidateRepository.findById(user.getCandidate().getId())
+                .orElseThrow(
+                    () -> new NoSuchObjectException(Candidate.class, user.getCandidate().getId()));
+            candidate.setChangePassword(false);
+        }
```

Comment:
You haven't saved the updated candidate.

I think you need

candidateRepository.save(candidate)

Reply:
done